test/PeerReview/Web/ServerSpec.hs

```diff
@@ -28,36 +29,79 @@ spec = before_ (wipeDb dbInfo) $ with app $ do
     it "responds with 404" $
       get "/non-existing-url" `shouldRespondWith` 404
 
-  describe "GET /" $
+  describe "GET root url" $
     it "responds with 200" $
       get "/" `shouldRespondWith` 200
 
-  describe "GET new review" $ do
-    context "when no submission can be reviewed" $
+  describe "POST create review" $ do
+    context "when new review can be found and created" $
+      it "responds with new review" $ do
+        let jsonBody     = [json| {userID: "user1"} |]
+            jsonResponse = [json|
+                { status: "waiting"
+                , submissionId: "1"
+                , score: 0
+                , reviewerId: "user1"
+                , comment: ""
+                , taskId: "task1"
+                , submissionContent: "wat"
+                , id: 1
+                } |]
+        post "/api/peer-reviews/create" jsonBody `shouldRespondWith` jsonResponse
+    context "when new review can't be found" $
       it "responds with an error message" $ do
-        let jsonError = [json|
-              { code: 1
-              , message: "No submissions to review."
-              }|]
-        get "/peer-reviews/new?email=test" `shouldRespondWith` jsonError
-    context "when submission can be reviewed" $
-      it "responds with an peer review" $ do
-        let jsonReview = [json|
-              { status: "waiting"
-              , submissionId: "1"
-              , score: 0
-              , reviewerId: "user1"
-              , comment: ""
-              }|]
-        get "/peer-reviews/new?email=user1" `shouldRespondWith` jsonReview
+        let jsonBody  = [json| {userID: "test"} |]
+        post "/api/peer-reviews/create" jsonBody `shouldRespondWith` 404
```

Comment:
Dunno if `404` is a correct HTTP code for response if JSON parsing fails. I'd go with `400`. http://stackoverflow.com/questions/6123425/rest-response-code-for-invalid-data